### PR TITLE
Get configured Rrdcached Socket from config

### PR DESCRIPTION
**Pull request by SourceDoctor**
Socket Configuration for RRDCached for Application "RRDCached" now fetched from Configuration

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/rrdcached-monitoring-custom-rrdcached-sock-dir/11358/12

**Comment by murrant** [comment]
Seems fine as there should be no situation where you are monitoring and running rrdcached and not using it in LibreNMS.

Maybe make sure it is a unix socket with a file_exists() check and not a tcp port on another system.

File: includes/polling/applications/rrdcached.inc.php [hunk 1]
```diff
@@ -41,14 +41,11 @@
...
-        if (file_exists('/var/run/rrdcached.sock')) {
-            $sock = fsockopen('unix:///var/run/rrdcached.sock');
-        } elseif (file_exists('/var/run/rrdcached/rrdcached.sock')) {
-            $sock = fsockopen('unix:///var/run/rrdcached/rrdcached.sock');
-        } elseif (file_exists('/run/rrdcached.sock')) {
-            $sock = fsockopen('unix:///run/rrdcached.sock');
-        } elseif (file_exists('/tmp/rrdcached.sock')) {
-            $sock = fsockopen('unix:///tmp/rrdcached.sock');
+        $socket = \LibreNMS\Config::get('rrdcached');
+        if ($substr($socket, 0, 7 ) == "unix://") {
```
Comment (TheGreatDoc) [inline includes/polling/applications/rrdcached.inc.php]: Is that `$substr` a misstype?

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-63-release-changelog-april-2020/11828/1

**Pull request merged**